tests/adapters/base.py

```diff
@@ -17,6 +19,44 @@ class _AdapterTester:
     def test_initialize(self, test_adapter):
         test_adapter._initialize_model()
 
+        # This comes from how test_adapter is initialised, we might want to get
+        # rid of this in future as it will interfere with defaults.
+        assert test_adapter._parameters.get_scalar_view(
+            ("rf2xco2",),
+            ("World",),
+            "W / m^2",
+        ).get() == 4
+
+        # Would it make sense for us to also pre-set the outputs with empty arrays?
+        # Then we could use these outputs to help us with separating model config
+        # from model timeseries. It may require a re-name, but that would also make
+        # it possible for the `initialize_model_input` and `initialize_run_parameters`
+        # methods to know which parameters they're meant to look at for the
+        # initialisation.
+        assert test_adapter._output.get_timeseries_view(
+            ("Emissions", "CO2"),
+            ("World",),
+            "GtCO2/yr",
+            [0, 1, 2],
+            ParameterType.AVERAGE_TIMESERIES,
+        ).is_empty
+
+        assert test_adapter._output.get_timeseries_view(
+            ("model specific para",),
+            ("model region",),
+            "model unit",
+            [0, 1],
+            ParameterType.POINT_TIMESERIES,
+        ).is_empty
+
+        # We could turn this into an abstract method so that implementers are forced
```

Comment:
yes, that should be left up to people implementing adapters

Reply:
ye but I guess my question is should it be forced? i.e. if we make this abstract, then they have to add a test like the below, if we don't then they can pass all our tests without ever writing a test that is specific to their model..

Reply:
@swillner shall I make this abstract so the adapters have to implement a test like it?